### PR TITLE
add github actions/goreleaser to release the lib

### PR DESCRIPTION
**Pull request by cpanato**
Fixes: #113 

rehearsal on my fork: https://github.com/cpanato/e2e-framework/runs/5220056082?check_suite_focus=true
release: https://github.com/cpanato/e2e-framework/releases/tag/v99.99.01

/assign @vladimirvivien

### CODE REVIEW

**Comment by cpanato** [comment]
if there is anything else to add let me know

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: release
+
+on:
+  create:
```
Comment (vladimirvivien) [inline .github/workflows/release.yml:4]: What's the difference between `create:` and `push:`

Comment (vladimirvivien) [inline .github/workflows/release.yml:4]: Should this be `push:` instead of `create:`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    tags:
+
```
Comment (vladimirvivien) [inline .github/workflows/release.yml:6]: Also, you may have to provide a tag name pattern for it to trigger (i.e. `v*` or `v*.*.**`)

**Review by vladimirvivien — APPROVED** [review]
Thanks for doing this! Left couple comments.

File: .github/workflows/release.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/release.yml:4]: both works for tags in this case, I can change for push, if you prefer

File: .github/workflows/release.yml (hunk 2 above)
Comment (cpanato) [reply .github/workflows/release.yml:6]: makes sense, i will update, thanks for the review

**Comment by vladimirvivien** [comment]
@cpanato I pressed `approved` by accident. Please see comments.

**Comment by cpanato** [comment]
put a hold @vladimirvivien then i can follow up :)

File: .github/workflows/release.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/release.yml]: updated to push

**Review by vladimirvivien — APPROVED** [review]
/lgtm

**Pull request merged**